src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -222,6 +222,11 @@ ZoneSet::MoveWindowIntoZoneByIndex(HWND window, HWND windowZone, int index) noex
         index = 0;
     }
 
+    while (auto zoneDrop = ZoneFromWindow(window))
+    {
+        zoneDrop->RemoveWindowFromZone(window, !IsZoomed(window));
+    }
+
     if (index < m_zones.size())
```

Comment:
This will never fail as we're checking it above.. Could remove this if

Reply:
You're right, I removed it